src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/DuplicateFinder.java

```diff
@@ -23,19 +24,14 @@ public DuplicateFinder(
         this.envelopeService = envelopeService;
     }
 
-    public List<BlobItem> findIn(String containerName) {
+    public List<Envelope> findIn(String containerName) {
         return storageClient
             .getBlobContainerClient(containerName)
             .listBlobs()
             .stream()
-            .filter(blob -> hasAlreadyBeenProcessed(blob.getName(), containerName))
+            .map(blob -> envelopeService.findEnvelope(blob.getName(), containerName))
+            .flatMap(Optional::stream)
```

Comment:
Getting rid of empty optionals (new way from Java 9)
https://www.baeldung.com/java-9-optional#the-stream-method)